SHARDING.md

```diff
@@ -0,0 +1,254 @@
+  role        = "roles/cloudkms.cryptoOperator"
+
+  member = "user:<username>@sigstore.dev"
+}
+```
+
+2. Run `terraform plan` and `terraform apply` using the GHA workflow.
+
+3. Using [tinkey](https://developers.google.com/tink/tinkey-overview#installation), create an encrypted Tink keyset:
+
+```
+tinkey create-keyset --key-template ED25519 --out enc-keyset.cfg --master-key-uri gcp-kms://projects/<project name>/locations/<region>/keyRings/<key-ring>/cryptoKeys/<key>
+```
+
+Save the output file `enc-keyset.cfg`, which will be used as a value in the Helm chart.
+
+4. Revoke IAM access by removing `google_kms_key_ring_iam_member`, and run `plan` and `apply`.
+
+## Creating K8s Resources
+
+Update `argocd/bootstrap-utilities/staging/values.yaml` or `argocd/bootstrap-utilities/production/values.yaml`
+to include the new shard. Under `rekor-tiles.shards`, copy the previous shard instantiation, and update
+the values accordingly based on the Terraform resources names. For `keyset`, copy the contents of
+`enc-keyset.cfg` (Since this value is encrypted, it's secure to include it in the config rather than
+use GCP Secrets).
+
+<!--- TODO: Add code pointers for values -->
+
+Merge, and wait for ArgoCD to create the resources and spin up the service.
+
+## Update Prober
+
+Update the prober configuration for
+[staging](https://github.com/sigstore/sigstore-probers/blob/8608ff2cbc6a9121f12c75cb01a18f32e1428195/.github/workflows/prober-staging.yml#L28)
+or
+[prod](https://github.com/sigstore/sigstore-probers/blob/8608ff2cbc6a9121f12c75cb01a18f32e1428195/.github/workflows/prober-prod.yml#L26)
+to specify the new shard name. Merge and kick off the workflow.
+
+<!--- TODO: Update prober to support testing against multiple Rekor shard URLs -->
+<!--- TODO: Update prober to test reads and writes separately, since frozen shards will still support reads -->
+
+## Perform Manual Testing
+
+Using Cosign, test signing and verification against this new shard. You will need to manually update the SigningConfig
+and TrustedRoot files and pass them via CLI flags since those files are pulled from TUF and will not yet have been updated.
+
+<!--- TODO: Update with example TrustedRoot and SigningConfig -->
+
+## Update TUF Verification Material
+
+Now that the shard is up and we've successfully signed and verified using it, we will
+update the TrustedRoot and SigningConfig files distributed via TUF so that clients
+can verify proofs from the shard and use the shard during signing.
+
+For TrustedRoot, the validity window for the shard should be from the point
+when the shard was spun up. For SigningConfig, the validity window should start
+after the TUF timestamp expires to guarantee all clients will have picked up
+the latest TrustedRoot. In practice for our TUF root with timestamp validity
+of one week, this means setting the validity window to be at least one week
+past when the TUF targets are signed.
+
+You will need to estimate two timestamps:
```

Comment:
I think we should have a well defined overlap period. maybe 1 day/week/month or whatever?

Having to guess this means people will ask around to get a recommendation instead of us just having this decision already made.